rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -41,11 +41,20 @@ void
 SequentialReader::open(
   const StorageOptions & storage_options, const ConverterOptions & converter_options)
 {
-  storage_ = storage_factory_->open_read_only(storage_options.uri, storage_options.storage_id);
+  file_uri_ = storage_options.uri;
+  storage_id_ = storage_options.storage_id;
+  storage_ = storage_factory_->open_read_only(file_uri_, storage_id_);
   if (!storage_) {
     throw std::runtime_error("No storage could be initialized. Abort");
   }
-  auto topics = storage_->get_metadata().topics_with_message_count;
+
+  BagMetadata bag_metadata = storage_->get_metadata();
```

Comment:
```suggestion
  const auto bag_metadata = storage_->get_metadata();
```

1/ if it can be const, make it const
2/ prefer auto to using types directly: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-auto